wallet/src/de/schildbach/wallet/util/DateExtensions.kt

```diff
@@ -1,14 +1,15 @@
 package de.schildbach.wallet.util
 
 import java.util.*
-import java.util.concurrent.TimeUnit
 
 fun Date.getDateAtHourZero(): Date {
-    val daysInMillis = TimeUnit.DAYS.toMillis(1)
-    val txDateCopy = Date(time)
-    val timeAtHourZero = time / daysInMillis * daysInMillis
-    txDateCopy.time = timeAtHourZero
-    return txDateCopy
```

Comment:
This code had a rounding issue where April 29 at 11:57 PM was converted to April 30 12 AM instead of April 29 12 AM.